src/components/Modal/Modal.jsx

```diff
@@ -0,0 +1,254 @@
+
+  // show/hide and animate the modal
+  useEffect(() => {
+    if (instance) {
+      const animate = instance?.$el.childNodes[1].animate;
+      if (show) {
+        instance?.show();
+        if (animate) instance?.$el.childNodes[1].animate?.(...getAnimationProps());
+      } else {
+        if (animate) {
+          const animation = instance?.$el.childNodes[1]?.animate(...getAnimationProps(true));
+          animation?.finished.then(() => instance?.hide());
+        } else {
+          instance?.hide();
+        }
+      }
+    }
+  }, [show, instance, getAnimationProps]);
+
+  // const hideModal = useCallback(() => {
+  //   instance.hide();
+  // }, [instance]);
+
+  const header = useMemo(() => {
+    const { id } = attr.title;
+    const header = React.Children.toArray(children).find(isModalHeader);
+    if (header) {
+      return cloneElement(header, { id, closeModal: onClose });
+    }
+
+    return (
+      <ModalHeader
+        title={title}
+        description={description}
+        closeModal={onClose}
+        id={id}
+        hideCloseButton={hideCloseButton}
+        closeButtonAriaLabel={closeButtonAriaLabel}
+      />
+    );
+  }, [children, attr, title, description, hideCloseButton, closeButtonAriaLabel]);
+
+  const content = useMemo(() => {
+    return (
+      React.Children.toArray(children).find(isModalContent) || (
+        <ModalContent>
+          {React.Children.toArray(children).filter(child => !isModalHeader(child) && !isModalFooter(child))}
+        </ModalContent>
+      )
+    );
+  }, [children]);
+
+  const footer = useMemo(() => {
+    return React.Children.toArray(children).find(isModalFooter) || null;
+  }, [children]);
+
+  const dialog = ReactDOM.createPortal(
+    <div
+      {...attr.container}
+      className={cx(styles.container, classNames.container)}
+      data-testid="monday-dialog-container"
+    >
+      {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events */}
+      <div
+        onClick={closeDialogIfNeeded}
+        className={cx(styles.overlay, classNames.overlay)}
+        data-testid="monday-modal-overlay"
+      />
+      <div
+        {...attr.dialog}
+        className={cx(styles.dialog, classNames.modal, {
+          [styles.default]: width === MODAL_WIDTH.DEFAULT,
+          [styles.full]: width === MODAL_WIDTH.FULL_WIDTH
+        })}
+      >
+        {header}
+        {content}
+        {footer}
+      </div>
+    </div>,
+    document.body
+  );
+
+  return <>{dialog}</>;
+};
+
+Modal.propTypes = {
+  /**
+   * Id of the modal, used internaly and for accessibility
+   */
+  id: PropTypes.string.isRequired,
+  /**
+   * Show/hide the Dialog
+   */
+  show: PropTypes.bool.isRequired,
+  /**
+   * Title for the modal, mandatory when ModalHeader isn't provided in children
+   */
+  title: function (props, propName, componentName) {
+    const { children, title } = props;
+    const hasHeaderComponent = React.Children.toArray(children).some(isModalHeader);
```

Comment:
nice!
is it worth it to maybe make this use the same logic as the [`header` const](https://github.com/mondaycom/monday-ui-react-core/pull/696/files#diff-1a35dbb21807dc16cc9af88fe302ff530a4426014b076ecbb8868e2b14bb773fR105)?

Reply:
@laviomri  it's almost the same, only difference is I'm using `some` instead of `find` because we need a boolean here instead of the actual bject